### ISSUE TITLE
RFC/RFE: manage.py command to rebuild trees

### ISSUE THREAD

**Issue by atodorov** [django-mptt/django-mptt#759]
I am experimenting with django-mptt for a legacy application which has in-house support for tree structures. It's a matter of updating the existing `parent` field to a `TreeForeignKey` and gradually start using the new API methods.

My question is what happens with the existing data on migrate ? Would tree structures be built automatically from mptt ? From what I can see not.

I am willing to contribute a manage.py command which will introspect models and rebuild the trees for all models that support them. Is this something you are interested in ?

**Comment by matthiask** [comment]
Hi @atodorov 

I'm personally more interested in other solutions for managing trees these days (see https://406.ch/writing/django-tree-queries/ for a write-up if what and why).

Have you seen the Node.objects.rebuild() tree manager method? It should be very useful for what you're doing except that it is only accessible though the shell, not as a management command right now.

I can see the potential usefulness of a manage.py rebuild_tree app_label.model_name management command, unfortunately not only for the initial tree building but also later.

**Comment by atodorov** [comment]
> Hi @atodorov
> 
> I'm personally more interested in other solutions for managing trees these days (see https://406.ch/writing/django-tree-queries/ for a write-up if what and why).

I've read your blog post and gave the tree_queries library a quick look. I like the simplicity but OTOH I need some more functionality that's currently present in django-mptt. Tough choice.

Just to be clear - you are still actively maintaining django-mptt, right ? (I can help with that too).

> 
> Have you seen the Node.objects.rebuild() tree manager method? It should be very useful for what you're doing except that it is only accessible though the shell, not as a management command right now.
> 

Yup, this is what I was planning to use and just put some Command glue around it.

> I can see the potential usefulness of a manage.py rebuild_tree app_label.model_name management command, unfortunately not only for the initial tree building but also later.

**Comment by matthiask** [comment]
Yes, I'm still (somewhat) actively maintaining django-mptt – not alone luckily, but additional help would be *very* welcome. 

I hope we will continue maintaining & extending the projects where we're using django-mptt at my employer for years to come.

**Comment by matthiask** [comment]
See #833 — I want to keep the project working for the time being but I'm not going to add (and maintain) new features. Thanks for your understanding.
